### DISCUSSION TITLE
navigation (skip, ffwd, rew,...) via accelerometer

### DISCUSSION THREAD

**Discussion by SeebM** [biologist79/ESPuino#42] [Ideas]
Hi,
I've discovered this project by chance, while I was planning to do more or less the same. So you spared me a lot of work. Thanks for that!
My initial idea was to reduce the necessary buttons by using an accelerometer (see sketch), which is of course derived from the original concept.
What are your thoughts on that? Would be happy to contribute, if I manage to get this set-up, which I cannot guarantee due to my limited experience in the field.
Best
Seeb
![TonieSchema](https://user-images.githubusercontent.com/76456388/102830775-66245f00-43ea-11eb-8ce3-4c56cfae78ef.png)

**Comment by biologist79** [comment]
Hehe, sounds interesting.
What's easy is to get the commands (next track, previous track...) sent. This is because there's a "universal" way using function trackControlToQueueSender(). Just put in your action there.

Biggest thing is to introduce a compile-switch that seperates between accelerometer and push-buttons. Don't know if probably GPIOs 36 and 39 (Sensor VP/VN) could be used to start an i2s-instance on them. Currently I don't have i2c running as per default.

One thing to mention: yesterday I introduced a bigger change into my devel branch (https://github.com/biologist79/Tonuino-ESP32-I2S/tree/devel). RFID is not yet working but I guess it will soon. So this will become the new project-layout (having setting-cfg-Files as per develboard; have a look at the src-folder).

**Reply by SeebM** [reply]
I did some experiments in the meantime and figured, that an accelerometer might not be the perfect solution for the problem. It is kind of hard to distinguish between normal movements and a real user command. However piezo elements, which are used for e-drums seem promising. Combined with an AD converter, this should work very simple. When my Tonuino is finished (woodwork, painting,..), I will try implement this feature. Might take some time though...

**Reply by biologist79** [reply]
Ok :-)
Regarding my previous proposal: VN/VP might not work as these ones are input-pins only.
However, do you have access to this one? https://discourse.voss.earth/t/vorstellung-marabox-bambus-3d-druck-neopixel-ring/4608
It's not an accelerometer but you don't need buttons as well.
